navigator_engine/pluggable_logic/data_loaders.py

```diff
@@ -87,3 +91,67 @@ def load_estimates_dataset(url_key: Hashable, auth_header_key: Hashable, engine:
             'data': {'completedTasks': []}
         }
     return data
+
+
+@register_loader
+def load_csv_from_zipped_resource(resource_type: str,
+                                  csv_filename_regex: str,
+                                  auth_header: str,
+                                  name: str,
+                                  engine: DecisionEngine) -> dict:
+
+    try:
+        data = load_estimates_dataset_resource(resource_type, auth_header, engine)
+        if not data[resource_type]['url'] \
+                and not data[resource_type]['auth_header'] \
+                and not data[resource_type]['data']:
+            raise ValueError(f"Empty resource {resource_type} in {data[resource_type]['url']}")
+    except IOError:
+        data = {
+            name: {
+                'data': None,
+                'auth_header': None,
+                'url': None
+            }
+        }
+        return data
+    except ValueError:
+        data[name] = {
+            'data': None,
+            'auth_header': data[resource_type].get('auth_header'),
+            'url': data[resource_type].get('url')
+        }
+        return data
```

Comment:
IOError is the wrong error for me to raise.  I will fix this in a later PR. Also, as I mentioned in my last comment I think the assumption should be that data[resource_type] could be empty. Not that an error gets thrown.  I think I need to fix this also in a later PR. Would suggest the following...

```suggestion
    data = load_estimates_dataset_resource(resource_type, auth_header, engine)
    if not data[resource_type]['data']:
        data[name] = {
            'data': None,
            'auth_header': None,
            'url': None
        }
        return data
```